pkg/controllers/node/ipam.go

```diff
@@ -77,8 +83,15 @@ func (c *NodeController) syncIPAMCleanup() error {
 
 			// Track nodes based on IP allocations.
 			if val, ok := attr.AttrSecondary[ipam.AttributeNode]; ok {
-				if _, ok := nodes[val]; !ok {
-					nodes[val] = []model.AllocationAttribute{}
+				kn := c.kubernetesNodeForCalico(val)
+				if kn == "" {
+					// If we can't determine a corresponding Kubernetes node for this allocation,
+					// we can't go any further. Move on to the next allocation.
+					logrus.WithField("calicoNode", val).Info("Could not find corresponding k8s node for allocation, skip")
+					continue
+				}
+				if _, ok := nodes[kn]; !ok {
+					nodes[kn] = []model.AllocationAttribute{}
```

Comment:
Not part of this PR, but.... firstSkip is always true - should be set to false below.